packages/design-system-mui/components/withController/index.tsx

```diff
@@ -0,0 +1,45 @@
+import { Controller } from 'react-hook-form'
+import { IControlledComponentProps } from './types'
+
+function withController(Component: any) {
```

Comment:
do u think we could move that `withController` to another package? Maybe `core` or we could create a `utils`

It being on the `design-system-mui` does not make much sense to me, since we could use that HOC for other purposes that does not include that design-system-mui itself (like, let's say I wanna create my own components using the react-hook-form but I wanna take advantage of the `withController`)

Reply:
Done, but now `design-system-mui` depends on `core`, which is probably ok, but not sure if we really want that.

Reply:
I think it's fine for now, for the future we're gonna have rethink core any way